cpp/tests/reductions/rank_tests.cpp

```diff
@@ -214,8 +214,8 @@ TYPED_TEST(TypedRankScanTest, StructsWithNullPushdown)
 
   // First, verify that if the structs column has only nulls, all output rows are ranked 1.
   {
-    struct_col->set_null_mask(
-      create_null_mask(12, cudf::mask_state::ALL_NULL));  // Null mask not pushed down to members.
+    // Null mask not pushed down to members.
+    struct_col->set_null_mask(create_null_mask(12, cudf::mask_state::ALL_NULL), struct_col->size());
```

Comment:
It seems fragile to have one length hardcoded as 12 and another written as a column size call. Should we be preferring the size call? Relatedly, can we do anything to not require/not provide this argument when the mask state is all nulls or no nulls?

Reply:
I don't see how we can make this easier other than providing some additional member functions. I don't think adding these is worth doing for this especially for the very rare all-nulls case.